File: src/test/java/no/utgdev/diy/ex5/Controller1.java
package no.utgdev.diy.ex5;

import no.utgdev.diy.annotations.Inject;
import no.utgdev.diy.annotations.Named;

public class Controller1 implements Controller {
    @Inject
    @Named("hello-dao")
    private HelloDao service;

    public String helloWorld() {
        return String.join(", ", service.hello());
    }
}

File: src/test/kotlin/no/utgdev/diy/ex5/Controller1.kt
package no.utgdev.diy.ex5

//REMOVE import no.utgdev.diy.annotations.Inject;
//REMOVE import no.utgdev.diy.annotations.Named;

class Controller1 : Controller {
    //REMOVE @Inject
    //REMOVE @Named("hello-dao")
    private val service: HelloDao? = null
    override fun helloWorld(): String {
        return java.lang.String.join(", ", service!!.hello())
    }
}